config/routes.php

```diff
@@ -101,6 +102,19 @@
 				return Panel::router($path);
 			}
 		],
+		// permalinks for page/file UUIDs
+		[
+			'pattern' => '@/(page|file)/(:all)',
+			'method'  => 'ALL',
+			'env'     => 'site',
+			'action'  => function (string $type, string $id) use ($kirby) {
+				// try to resolve to model, but only from UUID cache
+				$model = Uuid::for($type . '://' . $id)->resolve(true);
+				return $kirby
+					->response()
+					->redirect($model?->url() ?? 'error');
```

Comment:
I think it would be more consistent with the media routes if we render the error page directly. Untested, but I think this should work:

```suggestion

				if ($model !== null) {
					return $kirby
						->response()
						->redirect($model->url());
				}

				// render the error page
				return false;
```